src/components/PrivateRoute.tsx

```diff
@@ -1,27 +1,20 @@
 import { FunctionComponent } from "react";
 import { useSelector } from "react-redux";
 import { selectSession } from "modules/session";
-import { Redirect, Route, RouteProps } from "react-router-dom";
+import { Navigate, useLocation } from "react-router-dom";
 
-const PrivateRoute: FunctionComponent<RouteProps> = ({ children, ...rest }) => {
+type PrivateRouteProps = {
+  children: JSX.Element;
+};
+
+const PrivateRoute: FunctionComponent<PrivateRouteProps> = ({ children }) => {
   const { isLoggedIn } = useSelector(selectSession);
+  const location = useLocation();
 
-  return (
-    <Route
-      { ...rest }
-      render={ ({ location }) =>
-        isLoggedIn ? (
-          children
-        ) : (
-          <Redirect
-            to={ {
-              pathname: "/sign-up",
-              state: { from: location },
-            } }
-          />
-        )
-      }
-    />
+  return isLoggedIn ? (
+    children
+  ) : (
+    <Navigate to={ "/sign-up" } state={ { from: location } } replace />
```

Comment:
I haven't tested it, but I think something like this might be cleaner?

```
return (
  <Route 
    {...props} 
    element={(
      isLoggedIn
        ? element
        : <Navigate to="/sign-up" />
    )}
  />
)

```

Then, in the routes, you could just use

```
  <PrivateRoute 
    path="home" 
    element={<Home />}
  />

```

Reply:
Only a `<Route>` component can be a child of a `<Routes>` component. This means we can't use the`<PrivateRoute>` component as a direct child of `<Routes>`, which is why I had to change the current implementation. 

From the docs: 

"
[Refactor custom Routes
](https://reactrouter.com/docs/en/v6/upgrading/v5#refactor-custom-routes)Replace any elements inside a `<Switch>` that are not plain `<Route>` elements with a regular <Route>. This includes any `<PrivateRoute>`-style custom components.

You can [read more about the rationale behind this here](https://gist.github.com/mjackson/d54b40a094277b7afdd6b81f51a0393f), including some tips about how to use a `<Route render> `prop in v5 to achieve the same effect.
"

Reply:
Technically, it should still be a Route component. Does it enforce the component having to have the name "Route"?

Reply:
Read through the docs. Seems silly of them to not let you wrap a Route in your own component just because they thought people wouldn't like having to define the props.

Reply:
I agree. Kinda strange